packages/react/src/components/TearSheet/TearSheetWrapper.jsx

```diff
@@ -0,0 +1,180 @@
+    }
+  }, [containers, initialContainersWidth]);
+
+  useEffect(() => {
+    const greaterHeaderTopPosition = Math.max(
+      ...containers.map((c) => c.current.children[0]?.children[1]?.offsetTop ?? 0)
+    );
+
+    setAnimationClasses((prevState) => {
+      if (activeTearSheetIdx === null) {
+        return {
+          ...prevState,
+          overlay: [],
+        };
+      }
+      return {
+        ...Object.keys(prevState).reduce(
+          (acc, i) =>
+            i === 'overlay'
+              ? { ...acc, [i]: ['is-visible'] }
+              : i === `container${activeTearSheetIdx}`
+              ? { ...acc, [i]: [] }
+              : { ...acc, [i]: ['is-hidden'] },
+
+          {}
+        ),
+      };
+    });
+
+    const isWindowWidthSmallerThanContainer =
+      windowSize.width < initialContainersWidth + tearSheetConstants.DISTANCE_FROM_EACH_SIDE * 2;
+    const containerMaxWidth = windowSize?.width - tearSheetConstants.DISTANCE_FROM_EACH_SIDE * 2;
+    setContainersStyles(
+      containers.reduce((acc, c, idx) => {
+        const currentWidth = isWindowWidthSmallerThanContainer
+          ? containerMaxWidth - (activeTearSheetIdx - idx) * tearSheetConstants.WIDTH_DECREASE
+          : activeTearSheetIdx === null || activeTearSheetIdx <= idx
+          ? initialContainersWidth
+          : initialContainersWidth - (activeTearSheetIdx - idx) * tearSheetConstants.WIDTH_DECREASE;
+
+        const currentBottom =
+          activeTearSheetIdx !== null
+            ? activeTearSheetIdx < idx
+              ? 0 - windowSize.height
+              : activeTearSheetIdx === idx
+              ? 0
+              : (activeTearSheetIdx - idx) * tearSheetConstants.BOTTOM_POSITION_WHEN_HIDDEN
+            : 0 - windowSize.height;
+
+        return {
+          ...acc,
+          [`container${idx}`]: {
+            bottom: `${currentBottom}px`,
+            width: `${currentWidth}px`,
+            '--content-max-height': `${
+              windowSize?.height - greaterHeaderTopPosition - tearSheetConstants.DISTANCE_FROM_TOP
+            }px`,
+            '--content-max-width': `${containerMaxWidth}px`,
+          },
+        };
+      }, {})
+    );
+  }, [
+    containers,
+    activeTearSheetIdx,
+    initialContainersWidth,
+    windowSize.width,
+    windowSize.height,
+    windowSize,
+  ]);
+
+  const goToSheet = (idx) => setActiveTearSheetIdx(idx);
+
+  const closeAllTearSheets = () => {
+    if (onCloseAllTearSheets) {
+      onCloseAllTearSheets();
+      setActiveTearSheetIdx(null);
+    } else {
+      setActiveTearSheetIdx(null);
+    }
+  };
+
+  return (
+    <div
+      className={classnames(`${iotPrefix}--tear-sheet-wrapper`, className, {
+        ...animationClasses?.overlay?.reduce((acc, curr) => ({ ...acc, [curr]: isOpen }), {}),
+      })}
+      style={{ '--window-height': `${windowSize.height}px` }}
+      data-testid={`${iotPrefix}--tear-sheet-wrapper`}
+    >
+      {childrenArray.map((tearSheet, idx, arr) => {
```

Comment:
We may want to use [`React.Children.map`](https://reactjs.org/docs/react-api.html#reactchildrenmap)